### PR TITLE
Make encryption protocol configurable

### PR DESCRIPTION
**Pull request by ramindu90**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: component/src/main/java/io/siddhi/extension/store/mongodb/MongoDBEventTable.java [hunk 1]
```diff
@@ -140,6 +140,10 @@
                                 "the connection with TLS/SSL. false: Initiate the connection without TLS/SSL.",
                         defaultValue = "false",
                         possibleParameters = {"true", "false"}),
+                @SystemParameter(name = "encryptionProtocol",
+                        description = "Encryption protocol when `sslEnabled`.",
```
Comment (sajithshn) [inline component/src/main/java/io/siddhi/extension/store/mongodb/MongoDBEventTable.java]: Shall simply just call this "Encryption Protocol"

File: component/src/main/java/io/siddhi/extension/store/mongodb/util/MongoTableUtils.java [hunk 2]
```diff
@@ -646,7 +647,8 @@ private static SocketFactory extractSocketFactory(
         }
 
         try {
-            SSLContext sslContext = SSLContext.getInstance("SSL");
+            SSLContext sslContext =
+                    SSLContext.getInstance(configReader.readConfig(MongoTableConstants.ENCRYPTION_PROTOCOL, "TLSv1.2"));
```
Comment (sajithshn) [inline component/src/main/java/io/siddhi/extension/store/mongodb/util/MongoTableUtils.java]: Isn't the default value SSL according to doc annotation

**Review by sajithshn — APPROVED** [review]

File: component/src/main/java/io/siddhi/extension/store/mongodb/MongoDBEventTable.java (hunk 1 above)
Comment (ramindu90) [reply component/src/main/java/io/siddhi/extension/store/mongodb/MongoDBEventTable.java]: Fixed with https://github.com/siddhi-io/siddhi-store-mongodb/pull/61/commits/be04a98ace2f11b6ba59ad1333899db3d6025ac9

File: component/src/main/java/io/siddhi/extension/store/mongodb/util/MongoTableUtils.java (hunk 2 above)
Comment (ramindu90) [reply component/src/main/java/io/siddhi/extension/store/mongodb/util/MongoTableUtils.java]: fixed with https://github.com/siddhi-io/siddhi-store-mongodb/pull/61/commits/be04a98ace2f11b6ba59ad1333899db3d6025ac9

**Review by dnwick — APPROVED** [review]

**Pull request merged**